codebase2/codebase-sqlite/U/Codebase/Sqlite/NamedRef.hs

```diff
@@ -55,12 +55,15 @@ instance (FromRow ref) => FromRow (NamedRef ref) where
 -- Specifically, the scoped format adds the 'lastNameSegment' as well as adding a trailing '.' to the db format
 -- of both the namespace and reversed_name.
 --
--- Converts a NamedRef to SQLData of the form:
+-- This type has a ToRow instance of the form:
 -- [reversedName, namespace, lastNameSegment] <> ref fields...
-namedRefToScopedRow :: (ToRow ref) => NamedRef ref -> [SQLData]
-namedRefToScopedRow (NamedRef {reversedSegments = revSegments, ref}) =
-  toRow $ (SQLText reversedName, SQLText namespace, SQLText lastNameSegment) :. ref
-  where
-    reversedName = (Text.intercalate "." . toList $ revSegments) <> "."
-    namespace = (Text.intercalate "." . reverse . NEL.tail $ revSegments) <> "."
-    lastNameSegment = NEL.head revSegments
+newtype ScopedRow ref
```

Comment:
@ChrisPenner Heads up, this helper went away. I replaced `namedRefToScopedRow` with a newtype that emits the same `[SQLData]` in its `toRow` instance, because it's a little more straightforward to work into our QQ (actually -- heh -- I just realized you can interp a `[SQLData]` just fine -- but a newtype to house the instance seems a little cleaner)